### ISSUE TITLE
Typescript definition for equalTo() and notEqualTo() doesn't seem to acknowledge Array types as in the documentation.

### ISSUE THREAD

**Issue by swittk** [parse-community/Parse-SDK-JS#1260]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description

Following the guide at https://docs.parseplatform.org/js/guide/#queries-on-array-values
"you can find objects where the key’s array value"
````
// Find objects where the array in arrayKey contains 2.
query.equalTo("arrayKey", 2);
````

What actually happens is in Typescript definition, query.equalTo('arrayKey', <someobj>) would throw an error. (It expects the type of the key being searched match, which would be an array)

### Steps to reproduce

Try the Parse Queries on Array Values example in TypeScript

### Actual Outcome

````
Argument of type 'number' is not assignable to parameter of type 'number[]'.
````

### Expected Outcome

Typescript should not complain querying for values in arrays when using equalTo.

###  Environment

React Native 0.63

Client
- Parse JS SDK version: `2.18.0`

**Comment by fastrde** [comment]
Thank you for pointing this out. 
I am also using parse with Typescript and the definitions unfortunately don't match 100% of the code, but the definitions are not maintained here. It would be nice, when you could send a PR to https://github.com/DefinitelyTyped/DefinitelyTyped to fix this for us typescript users.

**Comment by swittk** [comment]
Hi there.
I currently have a stopgap solution for this problem, but I'm not sure this is the best way to approach this.
Currently the definition is like this
````
        equalTo<K extends (keyof T['attributes'] | keyof BaseAttributes)>(key: K, value: T['attributes'][K] | (T['attributes'][K] extends Object ? Pointer : never)): this;
````

To include array types, I've changed it to this for my use.
````
        equalTo<K extends (keyof T['attributes'] | keyof BaseAttributes)>(key: K, value: T['attributes'][K] | (T['attributes'][K] extends Object ? Pointer : never) | (T['attributes'][K] extends Array ? T['attributes'][K][0] : never)): this;
````

However, the checked type still allows either an array or an object, and I'm not sure if I should submit a PR with this typing since it's not truly correct. I'd be glad if someone could suggest a better solution.

**Comment by swittk** [comment]
Btw, just wondering, are there other functions that work on arrays other than equalTo and containsAll?

**Comment by swittk** [comment]
OK, I've submitted a PR to https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49818
I'll close this issue when that pull request gets received.

**Comment by swittk** [comment]
Fixed with release of @types/parse@2.81.1 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49818#issuecomment-736757432
